### PR TITLE
feat: ws for updating vote count and active connections

### LINKED ISSUE #159: Live count of how many users has voted
Suggested implementation:
Skjermbilde 2023-04-17 kl  12 50 30

Would be great using WebSockets for this for live updates

### PR DESCRIPTION
**Pull request by edalholt**
Closes #159 

Using WebSockets for updating vote count and active participants in real-time. 

When researching this issue I also noticed the basic plan in Azure is restricted to 350 concurrent connections.
For production the standard plan is therefore recommend.
https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits

Live count updated for every logged in organizer:
Skjermbilde 2023-08-08 kl  18 14 05

### CODE REVIEW

**Comment by Xtrah** [comment]
I have come to think of NTNUI/vote and NTNUI/opptak and regular pull requests. Perhaps we should remove the approval for PRs to dev (and maybe just commit directly) in these projects, and do a proper review from dev to main?

**Comment by edalholt** [comment]
Maybe, i am going to setup the production enviornment for vote soon,so we can consider it then. I also think it will be more difficult to understand what you are approving when it's consisting of 3-4+ pull requests

> I have come to think of NTNUI/vote and NTNUI/opptak and regular pull requests. Perhaps we should remove the approval for PRs to dev (and maybe just commit directly) in these projects, and do a proper review from dev to main?

**Comment by Xtrah** [comment]
Jumping between projects to review, I think it's a somewhat challenging already to understand what the different conventions are for each project. We should consider including a *How to test* in PR templates/descriptions, such as this one 😺

File: frontend/src/components/EditAssembly.tsx [hunk 1]
```diff
@@ -17,28 +17,53 @@ import {
...
+  // Request access to live assembly data for the given group when component is mounted.
+  useEffect(() => {
+    sendJsonMessage({ groupSlug: group.groupSlug });
+  }, []);
+
+  useEffect(() => {
+    // Update state every time the websocket receive a message.
+    if (lastMessage) {
+      const decodedMessage = JSON.parse(lastMessage.data);
+      console.log(decodedMessage);
```
Comment (LiviaValenti) [inline frontend/src/components/EditAssembly.tsx:54]: ✂️ ✨ 
```suggestion
```

File: frontend/src/components/EditAssembly.tsx [hunk 2]
```diff
@@ -17,28 +17,53 @@ import {
...
+      // User is is removed from current lobby if logged in on another device.
+      if (decodedMessage.participants != undefined) {
+        setParticipants(participants + parseInt(decodedMessage.participants));
+      }
+      if (decodedMessage.voteSubmitted != undefined) {
+        setSubmittedVotes(
+          submittedVotes + parseInt(decodedMessage.voteSubmitted)
+        );
+      }
```
Comment (LiviaValenti) [inline frontend/src/components/EditAssembly.tsx:63]: ✂️ ? 🤔
```suggestion
      if (decodedMessage.participants) {
        setParticipants(participants + parseInt(decodedMessage.participants));
      }
      if (decodedMessage.voteSubmitted) {
        setSubmittedVotes(
          submittedVotes + parseInt(decodedMessage.voteSubmitted)
        );
      }
```

File: frontend/src/components/EditAssembly.tsx [hunk 3]
```diff
@@ -49,10 +74,23 @@ export function EditAssembly(state: { group: UserDataGroupType }) {
...
+        if (vote.isActive) {
+          vote.options.forEach((option) => {
+            currentVotes += option.voteCount;
+          });
+        }
+      });
+      setSubmittedVotes(currentVotes);
     };
 
     fetch().catch(console.error);
```
Comment (LiviaValenti) [inline frontend/src/components/EditAssembly.tsx:96]: Should maybe be handled?

**Review by LiviaValenti — APPROVED** [review]
Super system 🤩 Nice job! Will test in dev as agreed upon 🤝

File: frontend/src/components/EditAssembly.tsx (hunk 3 above)
Comment (edalholt) [reply frontend/src/components/EditAssembly.tsx:95]: Yep, throw to sentry in the future? (#150)?

**Pull request merged**